develop-docs/sdk/expected-features/index.mdx

```diff
@@ -77,6 +77,31 @@ Backend SDKs (typically used in server applications) should have backpressure ma
 
 See <Link to="/sdk/telemetry/traces/backpressure/">Backpressure Management</Link> for details.
 
+## Debug Mode
+
+Every SDK must implement a debug mode, which is disabled by default. Users can enable it by setting the option `debug` to `true`. If the debug mode is enabled, the SDK prints out useful debugging information for two main purposes:
+
+	1. It helps users identify and fix SDK problems, such as misconfigurations or something going wrong with sending data to Sentry.
+	2. It can help Sentry SDK engineers investigate issues in Sentry SDKs, which can be done while developing or by asking a user to enable the debug mode and share the debug logs.
+
+As the log output can add unnecessary overhead, we advise users to refrain from enabling this in production. When using debug mode extensively for the second use case, we recommend adding the diagnostic level because users could easily miss error or fatal log messages.
```

Comment:
we also recently discussed that SDKs might log more critical/severe message (e.g. everything that would be `fatal`) independently of whether debug is set to true. should we mention this here as well?